sdk/communication/azure-communication-identity/src/test/java/com/azure/communication/identity/implementation/IdentityErrorConverterUnitTests.java

```diff
@@ -100,12 +98,9 @@ private CommunicationErrorResponseException setUpCommunicationResponseExceptionW
 
     private CommunicationErrorResponseException setUpCommunicationResponseExceptionWithAllProperties() {
         String value = "{\"code\":\"Error Code\",\"message\":\"Error Message\",\"target\":\"Error Target\",\"details\":[{\"code\":\"New Error Code\",\"message\":\"New Error Message\"}]}";
-        CommunicationError communicationError;
-        try {
-            communicationError = new ObjectMapper().readValue(value, CommunicationError.class);
-        } catch (JsonProcessingException e) {
-            throw new RuntimeException(e);
-        }
+        // Set up the response with all properties (no way to set Target?)
```

Comment:
@alzimmermsft is this is expected?

Reply:
Yes, but this is a case where you'd want to switch over the test code to using

```java
try (JsonReader jsonReader = JsonProviders.createReader(value)) {
    communicationError = CommunicationError.fromJson(jsonReader);
}
```

to have the stream-style deserialization method handle setting the values that are read-only (aka, cannot be set with a public API).